lua/telescope/make_entry.lua

```diff
@@ -858,16 +858,24 @@ function make_entry.gen_from_lsp_diagnostics(opts)
   opts = opts or {}
   opts.tail_path = utils.get_default(opts.tail_path, true)
 
-  local signs = {}
-  for _, v in pairs(lsp_type_diagnostic) do
-    signs[v] = vim.trim(vim.fn.sign_getdefined("LspDiagnosticsSign" .. v)[1].text)
+  local signs
+  if not opts.no_sign then
+    signs = {}
+    for _, v in pairs(lsp_type_diagnostic) do
+      local status, sign = pcall(
+        function() return vim.trim(vim.fn.sign_getdefined("LspDiagnosticsSign" .. v)[1].text) end)
```

Comment:
Wasn't an argument for `pcall` you can do this for example.
```lua
local get_table = function()
  return nil
  -- return { 'a' }
end

print((get_table() or {})[1])
```

Tell me what you like more, after that i can either merge or you update

Reply:
The thing is we basically just need a table so `x or {}` should just work

Reply:
I'll have to admit `pcall` is more simple here as otherwise there are two issues to fix:
1. The original issue empties out the text of the highlighting group with "" resulting in `SignTable[1].text` being nil - that was what I originally fixed
2. Your issue unbinds the diagnostic making`SignTable` nil, which is just as valid (maybe more) though probably less common than original issue

I'd quickly recommit though with comment as to why pcall is required. Sounds good?

Reply:
Sounds good